web-common/src/main/java/org/openmrs/eip/web/contoller/BaseRestController.java

```diff
@@ -52,6 +53,41 @@ public Object doGet(Long id) {
 		    "jpa:" + getName() + "?query=SELECT c FROM " + getName() + " c WHERE c.id = " + id, null, getClazz());
 	}
 	
+	public Map<String, Object> doSearchByPeriod(String startDate, String endDate) {
```

Comment:
By the way you are missing tests for your controller, can you add them please? It's good practice to always include tests for any code you write. Sorry I'm noticing this late

Reply:
Done